src/components/productrecommendation.fetchallpages.jsx

```diff
@@ -0,0 +1,163 @@
+    const { links } = this.state;
+    const link = links;
+    const arr = product.links;
+    for (let i = 0; i < arr.length; i += 1) {
+      if (arr[i].rel === 'element') {
+        link.push(arr[i]);
+      } else if (arr[i].rel === 'next') {
+        this.renderUrl(arr[i].uri);
+      }
+    }
+    this.setState({
+      links: link,
+    });
+  }
+
+  renderUrl(uri) {
+    const { links } = this.state;
+    const link = links;
+    login().then(() => {
+      fetch(`${Config.cortexApi.path + uri}`,
+        {
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+          },
+        })
+        .then(res => res.json())
+        .then((res) => {
+          const arr = res.links;
+          for (let i = 0; i < arr.length; i += 1) {
+            if (arr[i].rel === 'element') {
+              link.push(arr[i]);
+            } else if (arr[i].rel === 'next') {
+              this.renderUrl(arr[i].uri);
+            }
+          }
+          this.setState({
+            links: link,
+          });
+          return res;
+        });
+    });
+  }
+
+  static renderProducts(product, length, MaxItemsInOneCarouselView) {
+    const totalCount = product.length;
+    const maxViews = totalCount / MaxItemsInOneCarouselView;
+    const data = [];
+    for (let CurrentView = 0, CurrentItem = 0; CurrentView < maxViews; CurrentView += 1, CurrentItem += MaxItemsInOneCarouselView) {
+      if (CurrentView === 0) {
+        data.push(
+          <div className="carousel-item active" key={(CurrentItem)}>
+            <div className="row">
+              {Productrecommendationfetchallpages.renderCarouselView(CurrentItem, totalCount, product, MaxItemsInOneCarouselView)}
+            </div>
+          </div>,
+        );
+      } else {
+        data.push(
+          <div className="carousel-item " key={(CurrentItem + 1)}>
+            <div className="row">
+              {Productrecommendationfetchallpages.renderCarouselView(CurrentItem, totalCount, product, MaxItemsInOneCarouselView)}
+            </div>
+          </div>,
+        );
+      }
+    }
+    return (
+      <div>
+        {data}
+      </div>
+    );
+  }
+
+  static renderCarouselView(currentItem, totalCount, product, MaxItemsInOneCarouselView) {
+    /* Renders Each View in Carousel */
+    const data = [];
+    let itemCount = currentItem;
+    if (itemCount < totalCount && product[itemCount].rel === 'element') {
+      data.push(
+        <div className="col-6" key={`_${Math.random().toString(36).substr(2, 9)}`}>
+          <ProductListItemMain productUrl={product[itemCount].uri} />
+        </div>,
+      );
+    }
+    for (let i = 1; i < MaxItemsInOneCarouselView; i++) {
+      /* Copy this section to add more items in the same carousel view */
+      itemCount += 1;
+      if ((itemCount) < totalCount && product[(itemCount)].rel === 'element') {
```

Comment:
why is itemCount in brackets here? but not above. this is inconsistent with what you've already done

Reply:
👍